students/erica_edwards/lesson05/assignment/tests/test_database.py

```diff
@@ -0,0 +1,109 @@
+
+import unittest
+import os
+import logging
+from pymongo import MongoClient
+from pymongo import errors as pyerror
+from ..database import *
+
+logging.basicConfig(filename="db.log", filemode="w", level=logging.DEBUG)
+LOGGER = logging.getLogger(__name__)
+LOGGER.info("Started logger")
+
+class TestDatabase(unittest.TestCase):
+    """Tests for importing .csv files"""
+    def setUp(self):
+
```

Comment:
Try adding this code to your setup:

```
        """Clear the database created by running database.py directly"""
        mongo = MongoDBConnection()
        with mongo:
            db = mongo.connection.hp_norton_prototype
            db["product"].drop()
            db["customer"].drop()
            db["rental"].drop()
            db["customer_fail"].drop()
```

Your testing is currently passing because it inadvertently uses data that is created when you previously run the _main_ block of _database.py_. When you drop those tables, you will see this assertion error:

```
======================================================================
FAIL: test_show_rentals (assignment.tests.test_database.TestDatabase)
Test results returned show customers who are currently renting
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/students/erica_edwards/lesson05/assignment/tests/test_database.py", line 103, in test_show_rentals
    self.assertEqual(expected, result)
AssertionError: {'A003': {'address': '285 Frost Street', '[77 chars]69'}} != {}
+ {}
- {'A003': {'address': '285 Frost Street',
-           'email': 'frost@hotmail.com',
-           'name': 'Kelly Frost',
-           'phone_number': '2065874569'}}
```

Try to figure out how to get _test_database.py_ to pass without requiring a previous run of _database.py_